pkg/client/helper.go

```diff
@@ -245,8 +246,9 @@ func InClusterConfig() (*Config, error) {
 		Host:        "https://" + net.JoinHostPort(os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")),
 		Version:     "v1beta3",
 		BearerToken: string(token),
-		// TODO: package certs along with the token
-		Insecure: true,
+		TLSClientConfig: TLSClientConfig{
```

Comment:
I thought of one additional thing, sorry! Can you check the existence of this file and explicitly fail if it's not present? This is so someone compiling at head will figure out quickly that they're not secure when running on an old cluster.